bot/exts/filtering/filtering.py

```diff
@@ -66,6 +66,14 @@
 WEEKLY_REPORT_ISO_DAY = 3  # 1=Monday, 7=Sunday
 
 
+async def _extract_text_file_content(att: discord.Attachment) -> str:
+    """Extract up to the first 30 lines and first 2000 characters (whichever is shorter) of an attachment."""
+    file_encoding = re.search(r"charset=(\S+)", att.content_type).group(1)
+    file_lines: list[str] = (await att.read()).decode(encoding=file_encoding).splitlines()
```

Comment:
Can we split this into multiple lines? I don't like the inline await and the bunch of stuff chained after.